packages/@lwc/engine-core/src/framework/component.ts

```diff
@@ -14,47 +14,33 @@ import {
 import { VM, scheduleRehydration } from './vm';
 import { VNodes } from '../3rdparty/snabbdom/types';
 import { ReactiveObserver } from '../libs/mutation-tracker';
-import { LightningElementConstructor } from './base-lightning-element';
 import { Template, isUpdatingTemplate, getVMBeingRendered } from './template';
 
-export type ErrorCallback = (error: any, stack: string) => void;
-export interface ComponentInterface {
-    // TODO [#1291]: complete the entire interface used by the engine
-    setAttribute(attrName: string, value: any): void;
-}
-
-export interface ComponentConstructor extends LightningElementConstructor {
-    readonly name: string;
-    readonly labels?: string[];
```

Comment:
I don't recall name and labels usage, any insight here?

Reply:
@pmdartus what about these ones?

Reply:
Sorry I missed your comment. I did some archeology, and those 2 fields have been introduced with SnabbDOM v2 refactor: https://github.com/salesforce/lwc/pull/47. I don't recall us ever using those fields.